From fa3e89954070f7055f02c7c068f0ae578f3f2c28 Mon Sep 17 00:00:00 2001
From: kpandu549 <keerthana_pandurangan@comcast.com>
Date: Wed, 2 Mar 2022 09:53:17 +0000
Subject: [PATCH] REFPLTB-1602 : observing build errors for ccsp-hotspot
 component in 64bit

Reason for change: Observing below error on 64bit architecture
cosa_hotspot_dml.c:125:25: error: format '%d' expects argument of type 'int', but argument 4 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
|   125 |         CcspTraceError(("Invalid Client MAC address length %d, expected %zu\n", strlen(ret), sizeof(l_cMacAddr)));
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~
Test Procedure: build should be success on both 64bit and 32bit arch
Risks: Low

Signed-off-by: kpandu549 <keerthana_pandurangan@comcast.com>
Change-Id: Icaa79b78bfdb5a7d118138e63c32bfec1669bd18
---
 source/hotspotfd/cosa_hotspot_dml.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/hotspotfd/cosa_hotspot_dml.c b/source/hotspotfd/cosa_hotspot_dml.c
index 8b600ae..8e804aa 100644
--- a/source/hotspotfd/cosa_hotspot_dml.c
+++ b/source/hotspotfd/cosa_hotspot_dml.c
@@ -122,7 +122,7 @@ BOOL HotspotConnectedDevice_SetParamStringValue(ANSC_HANDLE hInsContext, char* P
         return FALSE;
     }
     if (strlen(ret) > sizeof(l_cMacAddr)) {
-        CcspTraceError(("Invalid Client MAC address length %d, expected %d\n", strlen(ret), sizeof(l_cMacAddr)));
+        CcspTraceError(("Invalid Client MAC address length %zu, expected %zu\n", strlen(ret), sizeof(l_cMacAddr)));
         return FALSE;
     }
     if(isValidMAC(ret) != 0)
-- 
2.28.0

